text/0207-context-scoped-attributes.md

```diff
@@ -0,0 +1,245 @@
+
+## Explanation
+
+The context-scoped attributes allows you to attach attributes to all telemetry signals emitted within a Context (or, following the usual rules of Context values, any child context thereof unless overridden).  Context-scoped attributes are normal attributes, which means you can use strings, integers, floating point numbers, booleans or arrays thereof, just like for span or resource attributes. Context-scoped attributes are associated with all telemetry signals emitted while the Context containing the Context-scoped attributes is active and are available to telemetry exporters. For spans, the context within which the span is started applies. Like other telemetry APIs, Context-scoped attributes are write-only for applications. You cannot query the currently set Context-scoped attributes, they are only available on the SDK level (e.g. to telemetry exporters, Samplers and SpanProcessors).
+
+Context-scoped attributes should be thought of equivalent to adding the attribute directly to each single telemetry item it applies to.
+
+### Comparing Context-scoped attributes to Baggage
+
+Context-scoped attributes and [baggage](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md) have two major commonalities:
+
+* Both are "bags of attributes"
+* Both are propagated identically in-process (Baggage also lives on the Context).
+
+However, the use cases are very different: Baggage is meant to transport app-level data along the path of execution, while execution-scoped attributes are annotations for all telemetry in the program execution "below" this point.
+
+This also explains the functional & API differences:
+
+* Most important point: Baggage is meant to be propagated; in fact, this is the main use case of baggage. While a propagator for Context-scoped attributes would be implementable, it would break the intended meaning by extending the scope of attributes to called services and would also raise many security concerns.
+* Baggage is both readable and writable for the application, while execution-scoped attributes, like all other telemetry attributes, are write-only.
+* Baggage only supports string values.
+* Baggage is not available to telemetry exporters (although e.g., a SpanProcessor could be used to change that), while that's the whole purpose of execution-scoped attributes.
+
+### Comparing Context-scoped attributes to Scope Attributes
+
+Context-scoped attributes and the recently added [Scope attributes](https://github.com/open-telemetry/oteps/pull/201) have these commonalities:
+
+* Both have "scope" in the name
+* Both apply a set of telemetry attributes to a set of telemetry items (those in their "scope")
+
+What is different is the scope. While Scope attributes apply to all items emitted by the same Tracer, Meter or LogEmitter (i.e., typically the same telemetry-implementing unit of code),
```

Comment:
>What is different is the scope.

I find it difficult to interpret this sentence. Perhaps delete it, since the rest reads fine without it.

>While Scope attributes apply to all items emitted by the same Tracer, Meter or LogEmitter (i.e., typically the same telemetry-implementing unit of code)

Perhaps also say that Scopes are defined at build-time as opposed to runtime nature of Context.

Reply:
I think the Instrumentation scope is _meant_ to be defined more or less statically, but technically, it is also a runtime API. But you are right that this point could be emphasized here.

Reply:
> What is different is the scope.

The scope in which the attributes apply is different between instrumentation scope and context scope. But yeah, it might be better to delete that.